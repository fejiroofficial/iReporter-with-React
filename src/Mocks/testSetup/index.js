/* eslint-disable import/no-extraneous-dependencies */
import '@babel/polyfill';
import React from 'react';
import { Provider } from 'react-redux';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import configMockStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';


export const mockState = {
  user: {
    token: '',
    isLoggedIn: false,
    loginStatus: false,
    message: '',
    isLoading: false,
    profile: {
      firstname: 'John',
      username: 'johnny',
      image: 'image.jpg',
      errorData: [],
    },
    resetPassword: {
      loading: false,
      message: '',
    }
  },
  bookmark: {
    loading: false,
    articles: [],
    message: '',
  },
  article: {
    newComments: {
      body: 'Hello world',
      id: 1,
      authorName: 'John',
      authorImage: 'image.jpg',
      date: 'July 4th 2018',
    },
    postingComment: false,
    commentMessage: 'Done',
    gettingComments: false,
    oldComments: [],
    hasMoreComments: false,
    commentPage: {
      current: 1,
      currentCount: 1,
      totalCount: 1,
    },
  },
  createArticle: {
    loading: false,
    message: false,
  },
  articlesReducer: {
    articles: []
  },
  articlesCategoryReducer: []
};


export const createMockStore = () => {
  const mockStore = configMockStore([thunk]);
  configure({ adapter: new Adapter() });
  return mockStore;
};

const setup = (component, initailState = mockState) => {
  const mockStore = configMockStore([thunk]);
  configure({ adapter: new Adapter() });
  const connectedWrapper = mount(
    <Provider store={mockStore(initailState)}>
      <BrowserRouter>
        {component}
      </BrowserRouter>
    </Provider>
  );
  return connectedWrapper;
};

export default setup;
