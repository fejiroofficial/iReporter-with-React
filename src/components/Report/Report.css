.Map {
    height: inherit;
}

  
.InfowindowContent .Title {
    font-weight: bold;
  }
  
.InfowindowContent {
    display: none;
  }
  
.Map .InfowindowContent {
    display: inline;
}

.PacCard {
    margin: 10px 10px 0 0;
    border-radius: 2px 0 0 2px;
    box-sizing: border-box;
    width: 299px;
    -moz-box-sizing: border-box;
    outline: none;
    box-shadow: 0 2px 6px rgba(0, 0, 0, 0.3);
    background-color: #fff;
    min-height: 467px;
}

.PacTitle {
    color: #fff;
    background-color: lightseagreen;
    font-size: 25px;
    font-weight: 500;
    padding: 6px 12px;
}

.PacContainer, .PacFile {
    padding-bottom: 12px;
    margin-right: 12px;
    margin-top: 24px;
}

.PacInput {
    background-color: #fff;
    font-family: inherit;
    font-size: 15px;
    font-weight: 300;
    margin-left: 12px;
    padding: 0 11px 0 13px;
    text-overflow: ellipsis;
    width: 276px;
    height: 45px;
    outline-color: lightseagreen;
}

.ReportBtn {
    height: 55px;
    width: 179px;
    margin-left: 55px;
    border-width: thin;
    border-radius: 8px;
    outline: none;
    border: none;
    margin-top: 2px;
    margin-bottom: 15px;
    padding-top: 2px;
    font-size: 20px;
    font-weight: bold;
    text-align: center;
    color: white;
    background-color: lightseagreen;
    cursor: pointer;
}

.SelectDrop {
    height: 37px;
    width: 276px;
    margin-top: 16px;
    margin-left: 12px;
    color: gray;
    padding-left: 8px;
    font-weight: bold;
    outline: none;
    font-family: inherit;
    font-size: 15px;
}

.UserComment {
    margin-left: 12px;
    width: 276px;
    margin-top: 25px;
    padding-left: 12px;
    padding-top: 12px;
    font-family: inherit;
    font-size: 15px;
    font-weight: lighter;
    outline-color: lightseagreen;
}

