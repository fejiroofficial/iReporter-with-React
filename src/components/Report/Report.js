import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import Classes from './Report.css';
import Aux from '../../hoc/Aux';

const report = (props) => (
  <Aux>
    <div className={Classes.PacCard}>
      <div>
        <div className={Classes.PacTitle}>
          iReporter form
        </div>
      </div>
      <textarea className={Classes.UserComment} placeholder="What's happening in your area?" cols="30" rows="5"></textarea>
      <div className={Classes.PacContainer}>
        <input className={Classes.PacInput} type="text" placeholder="Enter a location" />
      </div>
      <select id="styledSelect" className={Classes.SelectDrop}>
        <option value="red-flag">Red-flag</option>
        <option value="intervention">Intervention</option>
      </select>
      <div className={Classes.PacFile}>
        <input type="file" id="files" name="files" />
      </div>
      <button className={Classes.ReportBtn}>iReport</button>
      <div class="geo-display" id="geo-dis"></div>
    </div>
    <div className={Classes.Map}>
    </div>
    <div className={Classes.InfowindowContent}>
      <img src="" width="16" height="16" id="place-icon" />
      <span id="place-name" className={Classes.Title}></span><br />
      <span id="place-address"></span>
    </div>
  </Aux>
);

export default report;
