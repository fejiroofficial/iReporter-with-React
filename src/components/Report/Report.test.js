import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { createMockStore } from '../../Mocks/testSetup';
import Report from './Report';

createMockStore();

describe('<Report>', () => {
  it('should render without crashing', () => {
    const container = (
      <Report>
        <p>Hello</p>
      </Report>
    );
    const wrapper = shallow(container);
    expect(wrapper).toMatchSnapshot();
  });
});
