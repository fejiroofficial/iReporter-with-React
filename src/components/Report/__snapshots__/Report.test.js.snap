// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Report> should render without crashing 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
